operator/v1/types_console.go

```diff
@@ -143,8 +143,113 @@ type Capability struct {
+	// fileReference is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension for the dark or light theme so that the console serves the file with the correct MIME type.
+	// Recommended Masthead logo specifications:
+	// The Masthead logo must be at most 1 MiB in size and must have a maximum height of 60px and a maximum width of 200px.
+	// For the Masthead logo, the SVG file format is recommended, but other file formats are allowed if the browser supports them.
+	// Recommended Favicon specifications:
+	// The Favicon logo must be at most 1 MiB in size and must have a maximum height of 16px and a maximum width of 16px.
+	// For the Favicon logo, the PNG file format is recommended, but other file formats are allowed if the browser supports them.
+	// +required
+	FileReference LogoFileReference `json:"fileReference"`
+}
+
+// Logo defines a configuration based on theme modes for the console UI logo.
+type Logo struct {
+	// type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// type is a required field that allows values of Masthead and Favicon.
+	// When set to "Masthead", the logo will be used in the masthead and about modal of the console UI.
+	// When set to "Favicon", the logo will be used as the favicon of the console UI.
+	// +kubebuilder:validation:Enum="Masthead";"Favicon"
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes specifies the themes for the console UI logo.
+	// themes is a required field that allows a list of themes. Each item in the list must have a unique Type and a File field.
+	// Each theme determines whether the logo is for the dark or light theme of the console UI.
+	// There must be at least one entry and no more than 2 entries.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=map
+	// +listMapKey=mode
+	// +required
+	Themes []Theme `json:"themes"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
+// Ensure that Logos and CustomLogoFile cannot be set at the same time.
+// +kubebuilder:validation:XValidation:rule="!(has(self.logos) && has(self.customLogoFile))",message="Only one of logos or customLogoFile can be set."
 type ConsoleCustomization struct {
+	// logos is used to replace the OpenShift Masthead and Favicon logos in the console UI with custom logos.
+	// logos is an optional field that allows a list of logos.
+	// Only one of Logos or CustomLogoFile can be set at a time.
+	// If Logos is set, CustomLogoFile must be unset.
+	// When specified, there must be at least one entry and no more than 2 entries.
+	// Each item in the list must have a unique Type and a Themes field.
```

Comment:
"Each type must appear only once in the list"

You're mixing Go field names with the serialized versions here, all fields when described in godoc should be the seriali`ed format